node/core/approval-voting/src/approval_db/v2/mod.rs

```diff
@@ -0,0 +1,376 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Version 2 of the DB schema.
+
+use parity_scale_codec::{Decode, Encode};
+use polkadot_node_primitives::approval::{v1::DelayTranche, v2::AssignmentCertV2};
+use polkadot_node_subsystem::{SubsystemError, SubsystemResult};
+use polkadot_node_subsystem_util::database::{DBTransaction, Database};
+use polkadot_primitives::{
+	BlockNumber, CandidateHash, CandidateReceipt, CoreIndex, GroupIndex, Hash, SessionIndex,
+	ValidatorIndex, ValidatorSignature,
+};
+
+use sp_consensus_slots::Slot;
+
+use bitvec::{order::Lsb0 as BitOrderLsb0, vec::BitVec};
+use std::{collections::BTreeMap, sync::Arc};
+
+use crate::{
+	backend::{Backend, BackendWriteOp},
+	persisted_entries,
+};
+
+const STORED_BLOCKS_KEY: &[u8] = b"Approvals_StoredBlocks";
+
+mod migration_helpers;
+#[cfg(test)]
+pub mod tests;
+
+// DB migration support.
+pub use migration_helpers::{
```

Comment:
bikeshed: all the names have a redundant `migrate_approval_db_v1_to_v2` part that might be possible to remove if the module were just public.

Reply:
done